static/app/utils/api/useAggregatedQueryKeys.tsx

```diff
@@ -139,25 +149,24 @@ export default function useAggregatedQueryKeys<AggregatableQueryKey, Data>({
         );
       });
 
-      const promise = queryClient.fetchQuery({
-        queryKey: getQueryKey(queuedAggregatableBatch),
+      const queryKey = getQueryKey(queuedAggregatableBatch);
+      queryClient.fetchQuery({
+        queryKey,
         queryFn: fetchDataQuery(api),
       });
 
+      const observer = new QueryObserver(queryClient, {queryKey});
+      const unsubscribe = observer.subscribe(_result => {
+        queryClient.removeQueries({
+          queryKey: ['aggregate', cacheKey, key, 'inFlight'],
+          predicate: isQueryKeyInBatch,
+        });
+      });
+      subscriptions.current.push(unsubscribe);
+
       if (allQueuedQueries.length > queuedQueriesBatch.length) {
         fetchData();
       }
-
-      // When the promise resolves, it will trigger the cache subscription to fire
-      await promise;
-
-      queryClient.removeQueries({
-        queryKey: ['aggregate', cacheKey, key, 'inFlight'],
-        predicate: isQueryKeyInBatch,
-      });
-      queuedAggregatableBatch.forEach(queryKey => {
-        queryClient.setQueryData(['aggregate', cacheKey, key, 'done', queryKey], true);
```

Comment:
I'm a bit confused about this `done` key, I see it is set here but did anything read from this cache key?

Reply:
Ahh okay I see now, that makes sense